pymc3/sampling.py

```diff
@@ -1941,8 +1952,8 @@ def sample_prior_predictive(
     if data is None:
         raise AssertionError("No variables sampled: attempting to sample %s" % names)
 
-    prior = {}  # type: Dict[str, np.ndarray]
-    for var_name in vars:
+    prior: Dict[str, np.ndarray] = {}
+    for var_name in vars_:
```

Comment:
Previously, if the `if vars is None and var_names is None:` condition above was true, this iterator was over a `set`. 
Now `vars_` is no longer a set and may contain duplicates!!

Reply:
Good catch, thanks! OK, I've fixed this and have added a test which would fail if `vars_` wasn't a set